docs/bootstrap.json

```diff
@@ -0,0 +1,20 @@
+{
+  "ID": "bootstrap",
+  "Properties": {
+    "Resources": {
+      "resource_a": {
+        "Plugin": "resource-file",
+        "Type": "a",
+        "Properties": {
+          "Note": "Depends on {{.Resources.resource_b.ID}}"
```

Comment:
Another thing I would like to discuss is whether it makes sense to use a Golang template here as an *input* to the plugins.   It's fine to use a Golang template and we plan to do more with it in the future, but I am less certain about plugins accepting a template (which may not be a valid JSON at all) as input because plugins communicate using JSON-RPC currently.

This example here is a valid JSON because your reference is inside a string, but if I want to reference a complex object then something like

```json
{
    "id" : {{.Resources.somethingElse.NumericValue}},
    "value" : {{.Resources.another_resource.ComplexObject}}
}
```
can't be marshaled and unmarshaled as proper JSON.

In general, golang templates work fine *around the edges* prior to becoming the input to the plugins.  I think the best thing here is to introduce a struct into the document like:

```json
{
    "id" : { "ref" : "somethingElse.NumericValue" },
    "value" : { "ref" : "another_resource.ComplexObject"}
}
```
This is similar to the approach taken by Cloudformation and Azure templates.  This is a valid JSON and is easy to parse (no regex in your bootstrapper code).    It can also be sent as-is as input via JSON-RPC.  When it is evaluated, the references can be substituted with actual value and rendered as proper JSON.